substratevm/src/com.oracle.svm.agent.core/src/com/oracle/svm/agent/core/JNIHandleSetBase.java

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright (c) 2019, 2020, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package com.oracle.svm.agent.core;
+
+import static com.oracle.svm.core.util.VMError.guarantee;
+import static com.oracle.svm.jni.JNIObjectHandles.nullHandle;
+
+import java.util.Arrays;
+import java.util.concurrent.locks.ReentrantLock;
+
+import org.graalvm.nativeimage.c.type.CTypeConversion;
+
+import com.oracle.svm.jni.JNIObjectHandles;
+import com.oracle.svm.jni.nativeapi.JNIEnvironment;
+import com.oracle.svm.jni.nativeapi.JNIMethodId;
+import com.oracle.svm.jni.nativeapi.JNIObjectHandle;
+
+/**
+ * Helps with creation and management of JNI handles for JVMTI agents.
+ *
+ * A JVMTI agent must provide a subclass of this class. It should contain handles to classes and
+ * methods that are needed across different JVMTI calls (for example, a JNIObjectHandle to
+ * java/lang/Class).
+ *
+ * For JNI handles that are created in a JVMTI callback and that should survive a return from native
+ * code, a helper method {@link #newTrackedGlobalRef} is provided.
+ *
+ * @see com.oracle.svm.agent.core.JvmtiAgentBase
+ */
+public abstract class JNIHandleSetBase {
```

Comment:
I think this class is useful even without subclassing (although at least one instance is needed for the agent). I'd consider dropping the `Base` name suffix and change some methods from `protected` to `public`.

Reply:
I agree, I have dropped the base and changed the methods from protected to public.